tests/test_agpypeline.py

```diff
@@ -0,0 +1,487 @@
+    entry = entrypoint.__internal__()
+    assert entry.check_metadata_needed(configuration.Configuration()) is False
+
+
+def test_entrypoint_load_metadata_files():
+    """Tests entrypoint's load_metadata_files function by loading both an empty file and a metadata file
+    contained in the data folder and asserting that the respective results to the function call are the same
+    as these"""
+    entry = entrypoint.__internal__()
+    assert entry.load_metadata_files([]) == {'metadata': []}
+    orig = json.load(open("data/entrypoint_load_metadata_files.json"))
+    assert entry.load_metadata_files(["data/08f445ef-b8f9-421a-acf1-8b8c206c1bb8_metadata_cleaned.json"]) == orig
+
+
+def test_entrypoint_handle_check_continue_parse_continue_result():
+    """Tests entrypoint's handle_check_continue and parse_continue_result functions by passing in empty class
+     instances"""
+    entry = entrypoint.__internal__()
+    result = entry.handle_check_continue(algorithm.Algorithm(), environment.Environment(configuration.Configuration()),
+                                         {})
+    assert result == {}
+
+
+def test_entrypoint_handle_retrieve_files():
+    """Tests entrypoint's handle_retrieve_files function by passing in an empty class instances"""
+    entry = entrypoint.__internal__()
+    result = entry.handle_retrieve_files(environment.Environment(configuration.Configuration()), argparse.Namespace(),
+                                         [])
+    assert result is None
+
+
+def test_entrypoint_perform_processing():
+    """Tests entrypoint's perform_processing function by passing in empty class instances"""
+    namespace = argparse.Namespace()
+    namespace.file_list = []
+    namespace.working_space = []
+    entry = entrypoint.__internal__()
+    try:
+        entry.perform_processing(environment.Environment(configuration.Configuration()), algorithm.Algorithm(),
+                                 namespace, [])
+        assert True is False
+    except RuntimeError as error:
+        assert str(error) == "The Algorithm class method perform_process() must be overridden by a derived class"
+
+
+def test_entrypoint_handle_result():
+    """Tests entrypoint's handle_result function both by passing in empty parameters and by passing in a file path
+    to check for a result"""
+    entry = entrypoint.__internal__()
+    assert entry.handle_result({}) == {}
+    assert entry.handle_result({}, "", "") == {}
+    res = entry.handle_result({}, 'file', 'data/result.txt')
+    assert str(res) == open("data/result.txt").read()
+
+
+def test_entrypoint_add_parameters():
+    """Tests entrypoint's add_parameters function by passing in empty parameters"""
+    parser = argparse.ArgumentParser()
+    entrypoint.add_parameters(parser, algorithm.Algorithm(), environment.Environment(configuration.Configuration()))
+    assert str(parser.parse_args()) == "Namespace(debug=30, file_list=[], info=30, metadata=None," \
```

Comment:
I thought this was failing. How did you get it to work?

Reply:
It might have to do with the GitHub workflow running without/not using command-line parameters in the same way that I run it from a local python terminal. When I run pytest locally, the command itself gets sucked into argparse such as with the image I sent on Monday and then there are errors